src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Android.cs

```diff
@@ -128,19 +128,24 @@ private static void StopLoop()
 
         private static async Task PeriodicallyCheckIfNetworkChanged(CancellationToken cancellationToken)
         {
-            using var timer = new PeriodicTimer(s_timerInterval);
+            await Task.CompletedTask.ConfigureAwait(ConfigureAwaitOptions.ForceYielding);
 
+            var timer = new PeriodicTimer(s_timerInterval);
             try
             {
-                while (await timer.WaitForNextTickAsync(cancellationToken).ConfigureAwait(false)
-                    && !cancellationToken.IsCancellationRequested)
+                while (await timer.WaitForNextTickAsync(cancellationToken).ConfigureAwait(false) &&
+                       !cancellationToken.IsCancellationRequested)
                 {
                     CheckIfNetworkChanged();
                 }
             }
             catch (OperationCanceledException)
             {
             }
+            finally
+            {
+                timer.Dispose();
```

Comment:
Why is an explicit `try-finally` + `Dispose()` better here?

Reply:
At the IL level both will have two separate `try` blocks. The size difference seems to come from Roslyn adding a null check before calling `Dispose` (which it could elide by doing some basic code analysis) and also adding some weird useless code at the beginning of the try block (no idea where that comes from):
```
// sequence point: hidden
IL_0086: ldloc.0
IL_0087: ldc.i4.1
IL_0088: pop
IL_0089: pop
// sequence point: hidden
IL_008a: nop
```

Reply:
It is slightly smaller, but I changed it mainly because I was already editing this method, and I find fewer such constructs in the source easier to reason about.  A single try/catch/finally vs a using around a try/catch.  It's not a big deal either way.